### PR TITLE
Add taskplane extension

### PR DESCRIPTION
**Pull request by albingroen**
## Description

Added an extension for [Taskplane](http://taskplane.app) where you can browse and manage your tasks.

## Screencast

taskplane-1
taskplane-2
taskplane-3

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/taskplane/src/Task.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  }
+
+  return (
+    <Detail
+      navigationTitle={`#${task?.ref}`}
+      actions={
+        <ActionPanel>
+          <Action.OpenInBrowser
+            url={`${APP_URL}/${organizationId}/tasks?task=${task?.ref}`}
+            title="Open task"
```
Comment (pernielsentikaer) [inline extensions/taskplane/src/Task.tsx:90]:
```suggestion
            title="Open Task"
```

File: extensions/taskplane/src/Task.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            icon={Icon.Link}
+          />
+
+          <Action
+            onAction={() => {
+              handleUpdateTask({ isCompleted: !task.isCompleted });
+            }}
+            icon={{ source: task.isCompleted ? "check-circle.svg" : "check-circle-solid.svg", tintColor: COLORS.GREEN }}
+            title={`Mark as ${task.isCompleted ? "not completed" : "completed"}`}
```
Comment (pernielsentikaer) [inline extensions/taskplane/src/Task.tsx:99]:
```suggestion
            title={`Mark as ${task.isCompleted ? "Not Completed" : "Completed"}`}
```

File: extensions/taskplane/src/ChooseOrganization.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+              <ActionPanel>
+                <Action
+                  onAction={() => {
+                    push(
+                      <AuthContext.Provider value={{ token, setToken }}>
+                        <Tasks organizationId={organization.id} />
+                      </AuthContext.Provider>
+                    );
+                  }}
+                  title="Choose workspace"
```
Comment (pernielsentikaer) [inline extensions/taskplane/src/ChooseOrganization.tsx:54]:
```suggestion
                  title="Choose Workspace"
```

File: extensions/taskplane/src/Task.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          />
+
+          <Action
+            onAction={() => {
+              handleDeleteTask();
+            }}
+            shortcut={{ key: "backspace", modifiers: ["cmd"] }}
+            title="Delete task"
```
Comment (pernielsentikaer) [inline extensions/taskplane/src/Task.tsx:107]:
```suggestion
            title="Delete Task"
```

File: extensions/taskplane/src/sign-out.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            />
+          </ActionPanel>
+        }
+        title="Sign out"
+        subtitle="Sign out of your Taskplane account"
+        icon={Icon.XmarkCircle}
+      />
+    </List>
+  );
+}
```
Comment (pernielsentikaer) [inline extensions/taskplane/src/sign-out.tsx:28]:
```suggestion
import { List, ActionPanel, Action, showToast, Toast, Icon, useNavigation } from "@raycast/api";
import { logout } from "./lib";

const SignOut = async () => {
  logout();
};
export default SignOut;
```

File: extensions/taskplane/src/lib.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+});
+
+export const COLORS = {
+  GREEN: "#22C55E",
+  RED: "#EF4444",
+};
+
+export async function logout() {
+  LocalStorage.removeItem("x-taskplane-token");
+}
```
Comment (pernielsentikaer) [inline extensions/taskplane/src/lib.ts:20]:
```suggestion
export async function logout() {
  if (await confirmAlert({ title: "Are you sure you wnat to logout?" })) {
    LocalStorage.removeItem("x-taskplane-token");
    showToast(Toast.Style.Success, "Logged out successfully");
  }
}
```

File: extensions/taskplane/package.json [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      "subtitle": "Create a Taskplane task",
+      "description": "Create a Taskplane task",
+      "mode": "view"
+    },
+    {
+      "name": "sign-out",
+      "title": "Sign Out",
+      "subtitle": "Sign out of your Taskplane account",
+      "description": "Sign out of your Taskplane account",
+      "mode": "view"
```
Comment (pernielsentikaer) [inline extensions/taskplane/package.json:32]:
```suggestion
      "mode": "no-view"
```

File: extensions/taskplane/package.json [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  "author": "albingroen",
+  "categories": [
+    "Productivity"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "browse-tasks",
+      "title": "Browse Tasks",
+      "subtitle": "Browse your Taskplane tasks",
```
Comment (pernielsentikaer) [inline extensions/taskplane/package.json:16]:
```suggestion
      "subtitle": "Taskplane",
```

File: extensions/taskplane/package.json [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      "description": "Browse your Taskplane tasks",
+      "mode": "view"
+    },
+    {
+      "name": "create-task",
+      "title": "Create Task",
+      "subtitle": "Create a Taskplane task",
```
Comment (pernielsentikaer) [inline extensions/taskplane/package.json:23]:
```suggestion
      "subtitle": "Taskplane",
```

File: extensions/taskplane/package.json [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+      "description": "Create a Taskplane task",
+      "mode": "view"
+    },
+    {
+      "name": "sign-out",
+      "title": "Sign Out",
+      "subtitle": "Sign out of your Taskplane account",
```
Comment (pernielsentikaer) [inline extensions/taskplane/package.json:30]:
```suggestion
      "subtitle": "Taskplane",
```

File: extensions/taskplane/src/lib.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import { environment, LocalStorage } from "@raycast/api";
```
Comment (pernielsentikaer) [inline extensions/taskplane/src/lib.ts:1]:
```suggestion
import { confirmAlert, environment, LocalStorage, showToast, Toast } from "@raycast/api";
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @albingroen 👋 

Thanks for your contribution, awesome that you want to integrate into Raycast 🔥 

I left some suggestions and comments for you to address:

- I think It would be a lot better UX-wise and just have a no-view command for logout.
 
- The command `Browse Tasks` suffer from [Empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states)

When logging the `isLoading` variable I get this response:
```
08:15:25.662 true
08:15:25.665 false
08:15:25.665 true
08:15:25.675 false
08:15:25.747 false
```

It should only go from true to false 🙂 

- Corrected some naming conventions

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/taskplane/src/sign-out.tsx (hunk 5 above)
Comment (albingroen) [reply extensions/taskplane/src/sign-out.tsx:28]: It doesn't seem like `LocalStorage.removeItem` works in a `no-view` command?

Comment (pernielsentikaer) [reply extensions/taskplane/src/sign-out.tsx:28]: Let me check that

File: extensions/taskplane/package.json (hunk 8 above)
Comment (albingroen) [reply extensions/taskplane/package.json]: Fixed

File: extensions/taskplane/package.json (hunk 9 above)
Comment (albingroen) [reply extensions/taskplane/package.json]: Fixed

File: extensions/taskplane/package.json (hunk 10 above)
Comment (albingroen) [reply extensions/taskplane/package.json]: Fixed

File: extensions/taskplane/src/Task.tsx (hunk 4 above)
Comment (albingroen) [reply extensions/taskplane/src/Task.tsx]: Fixed

File: extensions/taskplane/src/ChooseOrganization.tsx (hunk 3 above)
Comment (albingroen) [reply extensions/taskplane/src/ChooseOrganization.tsx]: Fixed

File: extensions/taskplane/src/Task.tsx (hunk 2 above)
Comment (albingroen) [reply extensions/taskplane/src/Task.tsx]: Fixed

File: extensions/taskplane/src/Task.tsx (hunk 1 above)
Comment (albingroen) [reply extensions/taskplane/src/Task.tsx]: Fixed

File: extensions/taskplane/src/sign-out.tsx (hunk 5 above)
Comment (pernielsentikaer) [reply extensions/taskplane/src/sign-out.tsx:28]: Seems like there is a bug with `confirmAlert` - for now (until we fix it)

```
export async function logout() {
  await LocalStorage.removeItem("x-taskplane-token");
  showToast(Toast.Style.Success, "Logged out successfully");
}
```

Comment (albingroen) [reply extensions/taskplane/src/sign-out.tsx:28]: Yeah, I tried this too but it didn't work either. The toast never shows. Tried wrapping it in a try catch but nothing got logged there either.

Comment (pernielsentikaer) [reply extensions/taskplane/src/sign-out.tsx:28]: CleanShot 2022-07-08 at 09 34 36@2x
The toast should actually work, could you give it a go more?

Comment (albingroen) [reply extensions/taskplane/src/sign-out.tsx]: Fixed

File: extensions/taskplane/src/lib.ts (hunk 6 above)
Comment (albingroen) [reply extensions/taskplane/src/lib.ts]: Fixed

File: extensions/taskplane/package.json (hunk 7 above)
Comment (albingroen) [reply extensions/taskplane/package.json]: Fixed

File: extensions/taskplane/src/lib.ts (hunk 11 above)
Comment (albingroen) [reply extensions/taskplane/src/lib.ts]: Fixed

**Pull request merged**